### DISCUSSION TITLE
[CH4] 상수, 변수

### DISCUSSION THREAD

**Discussion by innerstella** [innerstella/ES2024#6] [Q&A] [answered]
상수와 변수. 그리고 상수 구분

**Comment by Lv1GoM** [comment, marked as answer]
### 상수와 변수
#### 변수 (Variable)
**선언**
(var, let, const)
**할당 (Assignment)** 
값을 변수에 할당할 수 있다.
**재할당 가능 (Reassignment)**
var와 let으로 선언된 변수는 값을 여러 번 할당하고 변경할 수 있다.
**스코프 (Scope)**
var는 함수 스코프를 가지고 있어 함수 내에서만 유효하고 함수 외부에서는 접근할 수 없다.
let은 블록 스코프를 가지고 있어 블록 내에서만 유효하며 블록 외부에서는 접근할 수 없다.

#### 상수 (Const)
**선언**
const 키워드를 사용하여 상수를 선언합니다.
**재할당 불가 (Reassignment)**
const로 선언된 상수는 선언 후에 값을 변경할 수 없다.
**스코프 (Scope)**
const도 블록 스코프를 가지고 있어 블록 내에서만 유효하다.

const는 상수의 재할당을 방지하지만, 객체 또는 배열과 같은 참조형 데이터의 내부 속성은 수정할 수 있다. 하지만 객체나 배열 자체를 다른 객체나 배열로 재할당하는 것은 불가능

#### 상수 구분
상수를 나타내는 변수명에는 보통 대문자만을 사용하며, 단어 사이를 언더스코어(_)로 연결. 이를 *"상수 케이스"* 또는 *"대문자 스네이크 케이스"* 라고 부름.
